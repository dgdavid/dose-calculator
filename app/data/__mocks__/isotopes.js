
const isotopes = {
  i1: {
    id: 'i1',
    name: 'Isotope 1',
    halfLife: 192.48,
    lambda: 0.0036,
  },
  i2: {
    id: 'i2',
    name: 'Isotope 2',
    halfLife: 6.01,
    lambda: 0.1153,
  },
};

export default isotopes;
