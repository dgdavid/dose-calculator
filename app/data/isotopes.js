const isotopes = {
  'iodine-131': {
    id: 'iodine-131',
    name: 'Iodine 131',
    halfLife: 192.48,
    lambda: 0.0036,
  },
  'technetium-99m': {
    id: 'technetium-99m',
    name: 'Technetium 99m',
    halfLife: 6.01,
    lambda: 0.1153,
  },
};

export default isotopes;
