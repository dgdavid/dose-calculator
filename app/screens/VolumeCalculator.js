import React, { Fragment } from 'react';
import { StyleSheet, Text } from 'react-native';

import { Row, Column, Label } from '../components/layout';
import { DateTimeInput, IsotopePicker, NumericInput } from '../components/fields';
import colors from '../config/colors';

export default () => (
  <Fragment>
    <Row size={0.2} />

    <Row size={0.9}>
      <Column>
        <Label text="Isotope" lineStyle={{ width: '50%' }} />
        <IsotopePicker prompt="Technetium 99m" />
      </Column>
    </Row>

    <Row size={0.9}>
      <Column>
        <Label text="Calibration Date" />
        <DateTimeInput />
      </Column>
    </Row>

    <Row size={0.2} />

    <Row size={2}>
      <Column style={styles.columnLeft}>
        <Label text="Initial activity" />
        <NumericInput unit={'mci'} />
      </Column>

      <Column>
        <Label text="Initial volume" />
        <NumericInput unit={'ml'} />
        <Text style={styles.warning}>{'\u26A0\uFE0F'} There is less volume than needed</Text>
      </Column>
    </Row>

    <Row size={2}>
      <Column style={styles.columnLeft}>
        <Label text="Desired activity" />
        <NumericInput unit={'mci'} />
      </Column>

      <Column>
        <Label text="Needed volume" />
        <NumericInput unit={'ml'} />
      </Column>
    </Row>

    <Row size={1.3}>
      <Column style={styles.columnLeft}>
        <Label text="Current activity" />
        <NumericInput
          unit={'mci'}
          style={styles.smallNumber}
          unitStyle={styles.smallUnit}
        />
      </Column>

      <Column>
        <Label text="Vial concentration" />
        <NumericInput
          editable={false}
          unit={'ml'}
          style={styles.smallNumber}
          unitStyle={styles.smallUnit}
        />
      </Column>
    </Row>

    <Row>
      <Column>
        <Label text="Â·" />
        <Text style={styles.info}>Information about the calculation</Text>
      </Column>
    </Row>
  </Fragment>
);

const styles = StyleSheet.create({
  columnLeft: {
    borderRightColor: colors.gray80,
    borderRightWidth: 0.5,
  },

  info: {
    padding: 10,
    alignSelf: 'center',
  },

  warning: {
    color: colors.orangered,
    fontSize: 10,
    alignSelf: 'center',
  },

  smallNumber: {
    fontSize: 20,
  },

  smallUnit: {
    fontSize: 10,
    padding: 0,
  },
});
