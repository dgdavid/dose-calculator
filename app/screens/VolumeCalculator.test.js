import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { VolumeCalculator } from './VolumeCalculator';
import isotopes from '../data/isotopes';

jest.mock('../data/isotopes');

jest.mock('../components/fields/DateTimeInput');

test('renders correctly', () => {
  const tree = renderer.create(<VolumeCalculator isotope={isotopes.i2} />).toJSON();

  expect(tree).toMatchSnapshot();
});
