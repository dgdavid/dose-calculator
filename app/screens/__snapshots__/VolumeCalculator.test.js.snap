// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Array [
        undefined,
        Object {
          "flex": 0.2,
          "flexDirection": "row",
        },
      ]
    }
  />,
  <View
    style={
      Array [
        undefined,
        Object {
          "flex": 0.9,
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginLeft": "5%",
              "width": "90%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "borderBottomColor": "#A9A9A9",
                "borderBottomWidth": 0.5,
                "marginTop": 10,
                "width": "90%",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "fontSize": 10,
                "marginLeft": 0,
                "marginTop": -7.5,
                "paddingLeft": 7.5,
                "paddingRight": 7.5,
              },
              undefined,
            ]
          }
        >
          ISOTOPE
        </Text>
      </View>
      <View
        style={
          Object {
            "borderWidth": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#696969",
              "fontSize": 30,
              "paddingTop": 5,
            }
          }
        >
          Technetium 99m
        </Text>
        <View
          style={
            Object {
              "height": 1000,
              "position": "absolute",
              "top": 0,
              "width": 1000,
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Technetium 99m",
                  "textColor": undefined,
                  "value": "technetium-99m",
                },
                Object {
                  "label": "Iodine 131",
                  "textColor": undefined,
                  "value": "iodine-131",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        undefined,
        Object {
          "flex": 0.9,
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginLeft": "5%",
              "width": "90%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "borderBottomColor": "#A9A9A9",
                "borderBottomWidth": 0.5,
                "marginTop": 10,
                "width": "90%",
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "fontSize": 10,
                "marginLeft": 0,
                "marginTop": -7.5,
                "paddingLeft": 7.5,
                "paddingRight": 7.5,
              },
              undefined,
            ]
          }
        >
          CALIBRATION DATE
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        undefined,
        Object {
          "flex": 0.2,
          "flexDirection": "row",
        },
      ]
    }
  />,
  <View
    style={
      Array [
        undefined,
        Object {
          "flex": 2,
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRightColor": "#CCC",
            "borderRightWidth": 0.5,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginLeft": "5%",
              "width": "90%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "borderBottomColor": "#A9A9A9",
                "borderBottomWidth": 0.5,
                "marginTop": 10,
                "width": "90%",
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "fontSize": 10,
                "marginLeft": 0,
                "marginTop": -7.5,
                "paddingLeft": 7.5,
                "paddingRight": 7.5,
              },
              undefined,
            ]
          }
        >
          INITIAL ACTIVITY
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          defaultValue="0"
          keyboardType="numeric"
          selectTextOnFocus={true}
          style={
            Array [
              Object {
                "color": "#696969",
                "fontSize": 40,
                "paddingBottom": 0,
                "textAlign": "center",
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={null}
        >
          mci
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginLeft": "5%",
              "width": "90%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "borderBottomColor": "#A9A9A9",
                "borderBottomWidth": 0.5,
                "marginTop": 10,
                "width": "90%",
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "fontSize": 10,
                "marginLeft": 0,
                "marginTop": -7.5,
                "paddingLeft": 7.5,
                "paddingRight": 7.5,
              },
              undefined,
            ]
          }
        >
          INITIAL VOLUME
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          defaultValue="0"
          keyboardType="numeric"
          selectTextOnFocus={true}
          style={
            Array [
              Object {
                "color": "#696969",
                "fontSize": 40,
                "paddingBottom": 0,
                "textAlign": "center",
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={null}
        >
          ml
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FF4500",
            "fontSize": 10,
          }
        }
      >
        ⚠️
         There is less volume than needed
      </Text>
    </View>
  </View>,
  <View
    style={
      Array [
        undefined,
        Object {
          "flex": 2,
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRightColor": "#CCC",
            "borderRightWidth": 0.5,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginLeft": "5%",
              "width": "90%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "borderBottomColor": "#A9A9A9",
                "borderBottomWidth": 0.5,
                "marginTop": 10,
                "width": "90%",
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "fontSize": 10,
                "marginLeft": 0,
                "marginTop": -7.5,
                "paddingLeft": 7.5,
                "paddingRight": 7.5,
              },
              undefined,
            ]
          }
        >
          DESIRED ACTIVITY
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          defaultValue="0"
          keyboardType="numeric"
          selectTextOnFocus={true}
          style={
            Array [
              Object {
                "color": "#696969",
                "fontSize": 40,
                "paddingBottom": 0,
                "textAlign": "center",
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={null}
        >
          mci
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginLeft": "5%",
              "width": "90%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "borderBottomColor": "#A9A9A9",
                "borderBottomWidth": 0.5,
                "marginTop": 10,
                "width": "90%",
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "fontSize": 10,
                "marginLeft": 0,
                "marginTop": -7.5,
                "paddingLeft": 7.5,
                "paddingRight": 7.5,
              },
              undefined,
            ]
          }
        >
          NEEDED VOLUME
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          defaultValue="0"
          keyboardType="numeric"
          selectTextOnFocus={true}
          style={
            Array [
              Object {
                "color": "#696969",
                "fontSize": 40,
                "paddingBottom": 0,
                "textAlign": "center",
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={null}
        >
          ml
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        undefined,
        Object {
          "flex": 1.3,
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRightColor": "#CCC",
            "borderRightWidth": 0.5,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginLeft": "5%",
              "width": "90%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "borderBottomColor": "#A9A9A9",
                "borderBottomWidth": 0.5,
                "marginTop": 10,
                "width": "90%",
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "fontSize": 10,
                "marginLeft": 0,
                "marginTop": -7.5,
                "paddingLeft": 7.5,
                "paddingRight": 7.5,
              },
              undefined,
            ]
          }
        >
          CURRENT ACTIVITY
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          defaultValue="0"
          keyboardType="numeric"
          selectTextOnFocus={true}
          style={
            Array [
              Object {
                "color": "#696969",
                "fontSize": 40,
                "paddingBottom": 0,
                "textAlign": "center",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "padding": 0,
            }
          }
        >
          mci
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginLeft": "5%",
              "width": "90%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "borderBottomColor": "#A9A9A9",
                "borderBottomWidth": 0.5,
                "marginTop": 10,
                "width": "90%",
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "fontSize": 10,
                "marginLeft": 0,
                "marginTop": -7.5,
                "paddingLeft": 7.5,
                "paddingRight": 7.5,
              },
              undefined,
            ]
          }
        >
          VIAL CONCENTRATION
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          defaultValue="0"
          editable={false}
          keyboardType="numeric"
          selectTextOnFocus={true}
          style={
            Array [
              Object {
                "color": "#696969",
                "fontSize": 40,
                "paddingBottom": 0,
                "textAlign": "center",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "padding": 0,
            }
          }
        >
          ml
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        undefined,
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "marginLeft": "5%",
              "width": "90%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "borderBottomColor": "#A9A9A9",
                "borderBottomWidth": 0.5,
                "marginTop": 10,
                "width": "90%",
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "fontSize": 10,
                "marginLeft": 0,
                "marginTop": -7.5,
                "paddingLeft": 7.5,
                "paddingRight": 7.5,
              },
              undefined,
            ]
          }
        >
          ·
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "padding": 10,
          }
        }
      >
        Information about the calculation
      </Text>
    </View>
  </View>,
]
`;
